### PR TITLE
🚸 A few more Vagrant features for ease of use.

### PR DESCRIPTION
**Pull request by bbaassssiiee**
Rather than using VirtualBox with manual setup, just install Ansible and Vagrant and run 'vagrant up'.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -1,11 +1,57 @@
...
+    end
+  end
+
+  # If new plugins installed, restart Vagrant process
+  if plugin_installed === true
+    exec "vagrant #{ARGV.join' '}"
+  end
+  config.vm.box_check_update = true
+  config.vm.graceful_halt_timeout=15
+  config.vbguest.auto_update = false
```
Comment (robertdebock) [inline Vagrantfile]: Thanks @bbaassssiiee.

I'm seeing (On Linux) this error when running `vagrant up`:
```
An error occurred while executing the action on the 'controlnode'
machine. Please handle this error then try again:

There are errors in the configuration of this machine. Please fix
the following errors and try again:

Vagrant:
* Unknown configuration section 'vbguest'.
```

File: Vagrantfile [hunk 2]
```diff
@@ -1,11 +1,57 @@
...
+  config.ssh.insert_key = false
+  config.ssh.forward_agent = true
+  config.hostmanager.enabled = true
```
Comment (robertdebock) [inline Vagrantfile]: I'm not 100% sure what the hostmanager does, but I assume it sets a hostname and fills /etc/hosts. The (included) role robertdebock.common does that too.

File: Vagrantfile [hunk 3]
```diff
@@ -1,11 +1,57 @@
...
+    web.vm.network "private_network", ip: "192.168.22.4", :netmask => "255.255.255.0",  auto_config: true
+    web.vm.provider :virtualbox do |vb|
+      vb.customize [
+          "modifyvm", :id,
+          "--name", "controlnode",
+      ]
+    end
+    web.vm.provision "ansible" do |ansible|
+    ansible.compatibility_mode = "2.0"
+    ansible.galaxy_role_file = "roles/requirements.yml"
```
Comment (robertdebock) [inline Vagrantfile:23]: The modules listed in there are installed in the playbook:
```
    - name: install required pip modules
      pip:
        name: "{{ item }}"
        state: present
      with_items:
        - pycodestyle
        - mock
        - voluptuous
```

**Review by robertdebock — CHANGES_REQUESTED** [review]
All in all the proposed changes do not work for me, but I've learned a bit, thanks @bbaassssiiee .

**Comment by bbaassssiiee** [comment]
Closing due to lack of activity.

**Pull request closed without merging**